### PR TITLE
feat: upgrade cli version

### PR DESCRIPTION
**Pull request by scefali**
The version of the CLI is quite out of date, this PR upgrades it to the latest which is 2.21.2.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
 import {getCLI} from './cli';
 import * as options from './options';
 import * as process from 'process';
 
+Sentry.init({
+  dsn: 'https://156a0eddcf407521a39833be862fc0e6@o1.ingest.sentry.io/4506112609943552',
+  // Performance Monitoring
+  tracesSampleRate: 1.0,
+});
+
```
Comment (mdtro) [inline src/main.ts:13]: Can we do this? A lot of repos use this action, wouldn't this collect data?
https://github.com/getsentry/action-release/network/dependents

File: Dockerfile [hunk 2]
```diff
@@ -21,11 +21,20 @@ RUN export YARN_CACHE_FOLDER="$(mktemp -d)" \
...
 # Copy the artifacts from `yarn build`
 COPY --from=builder /app/dist /action-release/dist/
 RUN chmod +x /action-release/dist/index.js
 
+# download a copy of sentry-cli
+# TODO: Figure out why we can't use the version that is in node modules
+RUN apk update && apk add curl
+RUN curl -sL https://sentry.io/get-cli/ | SENTRY_CLI_VERSION="2.21.2" sh
+RUN chmod +x /usr/local/bin/sentry-cli
+ENV SENTRY_BINARY_PATH=/usr/local/bin/sentry-cli
```
Comment (asottile-sentry) [inline Dockerfile:34]: this downloads the cli twice -- once in npm land and once here -- probably not what you want?  maybe you mean to execute `node_moduldes/.bin/sentry-cli ...` ?

**Comment by asottile-sentry** [comment]
there's a lot going on in this PR -- can you split it into several different patches to minimize risk?

probably separate out:
1. upgrade docker image version
2. upgrade npm packages
3. upgrade sentry-cli

File: Dockerfile (hunk 2 above)
Comment (scefali) [reply Dockerfile:34]: @asottile-sentry I tried this but seems like that doesn't work. I guess I'll keep trying.

**Pull request closed without merging**